doc/features/currencies.rst

```diff
@@ -49,6 +49,35 @@ The Bitcoin currencies provides a single currency: the Bitcoin. It uses XBT as i
     $currencies->contains(new Currency('USD')); // returns boolean whether USD is available in this repository (false)
     $currencies->subunitFor(new Currency('XBT')); // returns the subunit for the Bitcoin (8)
 
+CurrencyList
+-------------
+
+The CurrencyList class provides a way for a developer to build a custom currency repository.
+The class accepts an associative array indexed by the currency code
+and another associative array as the value containing `minorUnit` key and positive integer values.
+
+More specifically it must be an associative array. A currency code key must exists for all currencies and it must be a 
```

Comment:
I think this part is not necessary. Just mention in the previous section that an invalid array would end up in an exception being thrown.

Reply:
I believe it's fine now. Check it out :)